hfile.c

```diff
@@ -908,21 +1019,25 @@ static const struct hFILE_scheme_handler *find_scheme_handler(const char *s)
 
 hFILE *hopen(const char *fname, const char *mode, ...)
 {
+    hFILE *fp = NULL;
```

Comment:
Due to the route we arrived here, all this fp= assignment code is basically a no-change (my fault I think!).  I'll revert this bit back to the original prior to merging.